src/main/java/edu/unc/lib/boxc/migration/cdm/services/ArchivalDestinationsService.java

```diff
@@ -98,6 +105,44 @@ public Map<String, String> generateCollectionNumbersToPidMapping(DestinationMapp
         return mapCollNumToPid;
     }
 
+    /**
+     * Add archival collection mapping(s) to destination mappings file
+     * @param options
+     * @throws Exception
+     */
+    public void addArchivalCollectionMappings(DestinationMappingOptions options) throws Exception {
```

Comment:
There doesn't seem to be a test for this method at the moment, it would be good to add some. Some cases we should test would be:

- New file which generates at least one entry with a boxc pid
- New file which generates at least one entry without a boxc pid
- Existing file without force flag (throws an error, to prevent the user from accidentally overwriting whatever is already there)
- Existing file with force flag (overwrites the existing contents with the newly generated mapping)
- No field name is provided

Reply:
There's also one other bullet point on the ticket that I think needs to be addressed:
> If the destinations.csv file already exists, then overwrite it with a new file if the force flag is provided, otherwise throw an InvalidProjectStateException